themis/spec/symmetric-keys.md

```diff
@@ -12,4 +12,95 @@ Please come back later.
 Meanwhile, you may read our [key management guide](/themis/crypto-theory/key-management/).
 {{< /hint >}}
 
+Symmetric keys are used by [**Secure Cell**](../secure-cell/) cryptosystem.
+
+As of Themis 0.13 released in 2020,
+the following symmetric encryption algorithms are supported
+(in a multitude of modes):
+
+  - AES-128 (_deprecated_)
+  - AES-192
+  - AES-256 **(current default)**
+
+Each of the AES flavors has its own specific requirements for the key size.
+
+Secure Cell uses a _key derivation function_ (KDF)
+to transform user-provided key material into a symmetric key of suitable size.
+Therefore, keys used with Secure Cell do not have a particular format:
+they are just arbitrary non-empty byte strings.
+
+## Generating symmetric keys
+
+The recommended way to generate a symmetric key is to use a cryptographically secure pseudorandom number generator
+to obtain a sufficient number of random bytes, and that's your new symmetric key.
+
+{{< hint danger >}}
+**Warning:**
+It is crucial to use a *cryptographically secure* RNG so that the key is unpredictable.
+Do not use general-purpose random number generators – typically found in “math” libraries –
+unless they are explicitly designed for cryptographic purposes.
+{{< /hint >}}
+
+As of 2020,
+[NIST recommends](https://www.keylength.com/en/4/)
+symmetric keys to be at least 128 bits long for them to be future-proof.
+Themis currently generates 256-bit keys by default (i.e., 32 bytes).
+
+See also our [key management guide](/themis/crypto-theory/key-management/)
+for advice on securing the key after you have generated it.
+
+### Example
+
+Here is a snippet illustrating how to generate a good symmetric key:
+
+```go
+package main
+
+import (
+	"crypto/rand" // Use a cryptograpic RNG.
+	"fmt"
+)
+
+func main() {
+	// Generate keys of sufficient length.
+	const keyBytes = 32
+	key := make([]byte, keyBytes)
+	_, err := rand.Read(key)
+	if err != nil {
+		// Not handling RNG failure is a common critical error.
+		// If errors are ignored, you might return a predictable key,
+		// such as "all zeros" initial value of the byte array.
+		panic("failed to generate a key")
+	}
+	fmt.Printf("✅ symmetric key: %x\n", key)
+}
+```
+
 ## Passphrases
+
+Alternatively, Secure Cell can work with passphrases.
+From the cryptographic point of view, a passphrase is essentially a weird symmetric key.
```

Comment:
I would write not "weird", but "weak". But its just IMO

Reply:
> I would write not "weird", but "weak" [symmetric key].

Mmm... It's not like passphrases are weak _per se_: given a sufficient length and randomness, they can be effectively as good as the keys. I guess I wanted to communicate here that passphrases have lower entropy per byte than keys, that's why they are "worse".

I'll think of a better wording. At the moment, I'm out of ideas. @vixentael?

Reply:
Well, if that's about entropy then I've just written *that* as is, without calling passphrases weird. Please give it a look.